pkg/network/tracer/tracer_test.go

```diff
@@ -71,9 +71,7 @@ func TestMain(m *testing.M) {
 func setupTracer(t testing.TB, cfg *config.Config) *Tracer {
 	tr, err := NewTracer(cfg)
 	require.NoError(t, err)
-	t.Cleanup(func() {
```

Comment:
I don't think is a good idea because of the gc
As tr is local, I doubt there will have a reference when cleanups list is called (out of scope of this function)
so t.Cleanup may call Stop() with tr == nil

Reply:
Wouldn't `tr` not get gc'ed because there is a reference to `tr.Stop` in the cleanup list now?

Reply:
that way of cleanup is popular and works in other places, I just made sure we have a single way to teardown tr